api/v1alpha1/component_types.go

```diff
@@ -38,18 +39,11 @@ type GitSource struct {
 	DockerfileURL string `json:"dockerfileUrl,omitempty"`
 }
 
-// ComponentSource describes the Component source
-type ComponentSource struct {
-	ComponentSourceUnion `json:",inline"`
-}
-
-type ComponentSourceUnion struct {
-	// Git Source for a Component
-	GitSource *GitSource `json:"git,omitempty"`
-}
+// Component is the Schema for the components API. This is used to generate any k8s deployment resource
+type Component struct {
+	// For Component.Name as an alternative to using ComponentName
+	metav1.ObjectMeta `json:"metadata,omitempty"`
```

Comment:
Not sure why we want to introduce `metav1.ObjectMeta` here.. Any reason why Name is being replaced with this? ObjectMeta has more than Name that is perhaps not suited here

Reply:
Pulled that out and will only use Name and Namespace